verbit/streaming_client.py

```diff
@@ -193,45 +216,118 @@ def _connect_websocket(self, media_config: MediaConfig, response_types: Response
         # build websocket url
         ws_url = self.ws_url + self._get_ws_connect_query_string(media_config=media_config, response_types=response_types)
 
-        try:
+        # create WebSocket instance
+        self._ws_client = WebSocket(enable_multithread=True)
+
+        self._ws_client.timeout = 20.0
+        # ^ this kind of timeout handles 'changing networks' nicely
+        # but might be too short for hopeful TCP recovery...
+        # TBD + that's why a watchdog thread can sort-of do 'both'
+
```

Comment:
# ^ this kind of timeout handles 'changing networks' nicely
        # but might be too short for hopeful TCP recovery...
        # TBD + that's why a watchdog thread can sort-of do 'both'

Reply:
is this 20 seconds? it's forever!
what does this timeout control? only the establishment of the connection or also idleness etc.?

Reply:
Indeed let's discuss, it's a parameters of the WebSocket-client lib, which is passed to be a TPC level parameter AFAIK.
I've been experimenting with it, as written above- we might make a 'watchdog thread', with a much shorter timeout - that looks for an alternative route to the server, in-case it exists,
but in case it does now exist - leaving this timeout as high as possible ('None' -> default - system dependent, seems to be 60-sec or so on my devbox) - in case there's no alternative route - leaving the current WS open, will be the fastest recovery, in case the physical-layer reconnects.